src/test/java/com/mongodb/jdbc/MongoSQLResultSetTest.java

```diff
@@ -285,6 +320,175 @@ void testGetStringValues() throws Exception {
+        assertEquals(
+                ALL_MAX_KEY_COL_VAL, mongoSQLResultSetAllTypes.getString(ALL_MAX_KEY_COL_LABEL));
+
+        // Note that for "string" and "long", the ALL_X_COL_VAL must use extended json style-syntax in the
+        // Mock class, but the output for these types is not extended json style, by design.
+        assertEquals("str", mongoSQLResultSetAllTypes.getString(ALL_STRING_COL_LABEL));
+        assertEquals("5", mongoSQLResultSetAllTypes.getString(ALL_LONG_COL_LABEL));
+
+        // Note that the Java driver still outputs the legacy representation for DBPointer, as
+        // opposed to the new standard representation: { $dbPointer: { $ref: <namespace>, $id: <oid> } }.
+        // This is sufficient for our purposes, though.
+        assertEquals(
+                "{\"$ref\": \"db2\", \"$id\": " + ALL_OBJECT_ID_COL_VAL + "}",
+                mongoSQLResultSetAllTypes.getString(ALL_DB_POINTER_COL_LABEL));
+
+        // Note that getString() returns null for NULL and UNDEFINED BSON values
+        assertNull(mongoSQLResultSetAllTypes.getString(ALL_UNDEFINED_COL_LABEL));
+        assertNull(mongoSQLResultSetAllTypes.getString(ALL_NULL_COL_LABEL));
+    }
+
+    @Test
+    public void testGetObjectToStringAllTypes() throws Exception {
+        // Test getObject().toString() result for each BSON type.
+        assertEquals(
+                ALL_DOUBLE_COL_VAL,
+                mongoSQLResultSetAllTypes.getObject(ALL_DOUBLE_COL_LABEL).toString());
+        assertEquals(
+                ALL_OBJECT_COL_VAL,
+                mongoSQLResultSetAllTypes.getObject(ALL_OBJECT_COL_LABEL).toString());
+        assertEquals(
+                ALL_ARRAY_COL_VAL,
+                mongoSQLResultSetAllTypes.getObject(ALL_ARRAY_COL_LABEL).toString());
+        assertEquals(
+                ALL_OBJECT_ID_COL_VAL,
+                mongoSQLResultSetAllTypes.getObject(ALL_OBJECT_ID_COL_LABEL).toString());
+        assertEquals(
+                ALL_BOOL_COL_VAL,
+                mongoSQLResultSetAllTypes.getObject(ALL_BOOL_COL_LABEL).toString());
+        assertEquals(
+                ALL_REGEX_COL_VAL,
+                mongoSQLResultSetAllTypes.getObject(ALL_REGEX_COL_LABEL).toString());
+        assertEquals(
+                ALL_JAVASCRIPT_COL_VAL,
+                mongoSQLResultSetAllTypes.getObject(ALL_JAVASCRIPT_COL_LABEL).toString());
+        assertEquals(
+                ALL_SYMBOL_COL_VAL,
+                mongoSQLResultSetAllTypes.getObject(ALL_SYMBOL_COL_LABEL).toString());
+        assertEquals(
+                ALL_JAVASCRIPT_WITH_SCOPE_COL_VAL,
+                mongoSQLResultSetAllTypes
+                        .getObject(ALL_JAVASCRIPT_WITH_SCOPE_COL_LABEL)
+                        .toString());
+        assertEquals(
+                ALL_INT_COL_VAL, mongoSQLResultSetAllTypes.getObject(ALL_INT_COL_LABEL).toString());
+        assertEquals(
```

Comment:
This should not pass because mongoSQLResultSetAllTypes.getObject(ALL_TIMESTAMP_COL_LABEL) will return a java.sql.Timestamp and toString on it should return yyyy-mm-dd hh:mm:ss.fffffffff, not an extended JSON value like {"$timestamp": {"t": 1412180887, "i": 1}}.
I am a little surprised Evergreen is happy. Could you double-check? I am not sure what I missed here ?🧐

Reply:
Discussed offline: bson timestamp != bson date